cryptographic_estimators/MREstimator/MRAlgorithms/support_minors.py

```diff
@@ -0,0 +1,261 @@
+        lv = parameters["lv"]
+        b = parameters["b"]
+        nprime = parameters["nprime"]
+        q, m, _, k_reduced, r =  self.get_problem_parameters_reduced(a, lv)
+        dim = self._dimension(q, nprime, k_reduced +  1, r, b) - 1
+        exp = self._expected_dimension_of_support_minors_equations(q, m, nprime, k_reduced + 1, r, b)
+        return dim < exp
+
+    def _valid_choices(self):
+        """
+        Generator which yields on each call a new set of valid parameters based on the `_parameter_ranges` and already
+        set parameters in `_optimal_parameters`
+
+        """
+        new_ranges = self._fix_ranges_for_already_set_parameters()
+        _ = new_ranges.pop(MR_VARIANT)
+        indices = {i: new_ranges[i]["min"] for i in new_ranges}
+        variant = MR_STRASSEN
+        keys = [i for i in indices]
+        stop = False
+        while not stop:
+            if not self._are_parameters_invalid(indices):
+                aux = indices.copy()
+                aux.update({MR_VARIANT: variant})
+                yield aux
+            indices[next(iter(indices))] += 1
+            for i in range(len(keys)):
+                if indices[keys[i]] > new_ranges[keys[i]]["max"]:
+                    indices[keys[i]] = new_ranges[keys[i]]["min"]
+                    if i != len(keys) - 1:
+                        indices[keys[i + 1]] += 1
+                    elif i == len(keys) - 1 and MR_VARIANT == MR_STRASSEN:
+                        variant = MR_BLOCK_WIEDEMANN
+                        indices = {i: new_ranges[i]["min"] for i in new_ranges}
+                    else:
+                        stop = True
+                else:
+                    break
+
+    def _sm_time_complexity_helper_(self, q, K, r, nprime, b, variant):
+        if variant == MR_BLOCK_WIEDEMANN:
+            time = _bw_complexity_(row_density=K * (r + 1), ncols=self._dimension(q, nprime, K, r, b))
+
+        else:
+            time = _strassen_complexity_(rank=self._dimension(q, nprime, K, r, b) - 1,
+                                                            ncols=self._dimension(q, nprime, K, r, b))
+        return time
+
+    def _sm_memory_complexity_helper_(self, q, K, r, nprime, b, variant):
+        ncols = self._dimension(q, nprime, K, r, b)
+        if variant == MR_BLOCK_WIEDEMANN:
+            memory = log2(log2(q)) + log2(2 * ncols)
+
+        else:
+            memory = 2 * log2(ncols)
+        return memory
+
+    def _compute_time_complexity(self, parameters: dict):
+        """
+        Return the time complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        """
+        a = parameters[MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS]
+        lv = parameters[MR_NUMBER_OF_COEFFICIENTS_TO_GUESS]
+        nprime = parameters[MR_REDUCED_NUMBER_OF_COLUMNS]
+        b = parameters[MR_LINEAR_VARIABLES_DEGREE]
+        variant = parameters[MR_VARIANT]
```

Comment:
this is inconsistent with the previous usage of "a", "lv" etc. instead of MR_NUMBER_OF_KERNEL_VECTORS_TO_GUESS...

I see the advantage of the here used way of defining a, lv, nprime etc. over the SimpleNamespaces, as it reminds the user / developer what a, lv etc. was. In that case I would be fine with either variant.